ai/neuronal_network/dataset_preparation.py

```diff
@@ -0,0 +1,103 @@
+import json
+import numpy as np
+from keras.layers import Input, Dense, Flatten
+from keras.models import Model
+
+json_data = json.load(open('./trainning_data/trainning_data.json'))
+armies_length = 0
+nb_stats = 12
+
+
+def format_json_match(match):
+    first_player_units = []
+    second_player_units = []
+
+    for unit in match['first_player']['units']:
+        if unit is not None:
+            first_player_units.append(list(unit['stat'].values()))
+    for unit in match['second_player']['units']:
+        if unit is not None:
+            second_player_units.append(list(unit['stat'].values()))
+
+    if len(first_player_units) == 0 or len(second_player_units) == 0:
+        return None
+    return {
+        'first_player_units': first_player_units,
+        'second_player_units': second_player_units,
+        'first_player_score': match['first_player']['score'],
+        'second_player_score': match['second_player']['score'],
+        'first_player_cost': match['first_player']['cost'],
+        'second_player_cost': match['second_player']['cost']
+    }
+
+
+def purge_data(data):
+    # Remove None
+    data = list(filter(lambda x: x is not None, data))
+    # Remove empty units
+    data = list(filter(lambda x: len(x['first_player_units']) != 0, data))
+    data = list(filter(lambda x: len(x['second_player_units']) != 0, data))
+    # Remove units with missing stats
+    data = list(filter(lambda x: not any(
+        len(x) != nb_stats for x in x['first_player_units']), data))
+    data = list(filter(lambda x: not any(
+        len(x) != nb_stats for x in x['second_player_units']), data))
+    # Remove matchs with score != 20
+    data = list(filter(
+        lambda x: x['first_player_score'] + x['second_player_score'] == 20, data))
```

Comment:
For the matches were only one player has a score why not try to complete to 20 to keep more matches ?

Reply:
I'm not sure that I understand, what do you mean by only one player has a score ?

Reply:
oh I see what you mean,
Maybe we  can do this but I'm scared that some more information are missing on player 2 if his score is not reported. That's the kind of thing we should do if we really really lake of trainning data. And for the moment this filter does not deduce that much matches